go/types/set_iterator.go

```diff
@@ -32,48 +32,42 @@ type SetIterator interface {
 }
 
 type setIterator struct {
-	s      Set
-	cursor *sequenceCursor
+	s            Set
+	cursor       *sequenceCursor
+	currentValue Value
 }
 
 func (si *setIterator) Next() Value {
 	if si.cursor == nil {
-		si.cursor = newCursorAt(si.s.seq, emptyKey, false, false, false)
-	} else {
-		si.cursor.advance()
+		d.Panic("Cannot use a nil SetIterator")
```

Comment:
Could you remove this condition and the panic, I don't think it's useful. Also in map iterator. I made that comment but Aaron didn't see it I suppose.